docs/enhancements/cancel-day-2-host-installation.md

```diff
@@ -0,0 +1,177 @@
+
+#### Story 2
+
+As a cluster admin using Hypershift, I want to use the autoscaler.
+When the scaling happens and hosts are bound/unbound from the cluster, the hosts should bind and unbind 
+easily from the cluster.
+
+#### Story 3
+
+As a cluster admin, if I install a day-2 host to a cluster, and I see the host installation fails from
+inside the host. I should be able to cancel the installation once it starts and quickly reset the host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There is an already existing cancel host installation API which maps to the unbind and reclaim stages.
+This is currently only being used when a day-1 cluster installation is cancelled. 
+
+#### Detection of Cancellation
+
+When an Agent's cluster deployment reference is removed and the host is in the middle of installation,
+the controller will consider that a trigger to cancel the installation.
+
+Currently, there is a webhook that prevents the cluster reference from being removed
+when a host is in the middle of installation. The proposal for this enhancement is to remove
+that webhook to unblock the removal of the cluster reference.
+
+#### Cancelling Installation
+
+The host should transition to cancelled state.
+
+The `stop-installation` command is sent to a host that is in the middle of installing.
+The command will be sent with retries in case there is a scenario such as the host just starting rebooting.
+
+However, if the command doesn't receive a successful reply, the controller will assume
+the host is unreachable and transition the host to a pending-user-action state.
+
+#### Post Cancellation
+
+##### Successful Cancellation
+
+If a host cancellation was successful, the host will be transitioned to unbinding then reclaiming.
+
+During reclaim, the host should be rebooted with the discovery ISO and its state should be reset to
+discovering once the host tries to register again. This already exists with the current reclaim flow
+
+##### Failed Cancellation
+
+If the host cancellation fails due to the host being unreachable or timing out, then the
+host will transition to a pending-user-action state.
+
+From here, the user must take action by deleting the Agent CR to "reset" the host
+and rebooting the host manually from the discovery ISO.
+
+### Risks and Mitigations
+
+#### Risk 1: Unintended consequences
+
+Host installation cancellation is a completely new flow that could have unintended consequences.
+For example, it might not be able to be cancelled during a certain part of installation, or the
+host might end up in a state that's unreachable rendering it stuck.
+
+##### Mitigation
+
+The way to mitigate this is to fully explore and ensure all paths into and out of this state are covered.
+When testing out this feature, we should test cancelling the installation at different times and phases
+to ensure that cancellation is possible.
```

Comment:
I'd suggest to also provide a documented and supported manual procedure to return the host to the pool.